witness/golang/omniwitness/usbarmory/storage/internal.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build usbarmory
+// +build usbarmory
+
+// Package storage provides support for using the SD/eMMC storage provided
+// by the USB Armory.
+package storage
+
+import "github.com/usbarmory/tamago/soc/imx6/usdhc"
+
+var (
+	// MaxTransfer is the largest transfer we'll attempt.
+	// If we're asked to read or write more data than can fit into available DMA memeory
+	// we'll had a bad time, so we'll chunk into requests of at most MaxTransfer bytes.
+	MaxTransfer = 32 * 1024
+)
+
+// Device allows writing to one of the USB Armory storage peripherals, hiding some
+// of the sharp edges around DMA etc.
+type Device struct {
+	Card *usdhc.USDHC
+}
+
+// BlockSize returns the size in bytes of the each block in the underlying storage.
+func (d *Device) BlockSize() uint {
+	return uint(d.Card.Info().BlockSize)
+}
+
+// WriteBlocks writes the data in b to the device blocks starting at the given block address.
+// If the final block to be written is partial, it will be padded with zeroes to ensure that
+// full blocks are written.
+func (d *Device) WriteBlocks(lba uint, b []byte) error {
+	if len(b) == 0 {
+		return nil
+	}
+	ci := d.Card.Info()
```

Comment:
In both this and the Read method, I think it would be clearer to do `blockSize := d.BlockSize()` and use that. It doesn't look like `ci` is used for anything other than querying that.

Reply:
done